control-base/src/main/java/fi/nls/oskari/control/data/LayerAndGroupOrderHandler.java

```diff
@@ -0,0 +1,172 @@
+package fi.nls.oskari.control.data;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import fi.mml.map.mapwindow.service.db.OskariMapLayerGroupService;
+import fi.mml.map.mapwindow.service.db.OskariMapLayerGroupServiceIbatisImpl;
+import fi.nls.oskari.annotation.OskariActionRoute;
+import fi.nls.oskari.control.ActionDeniedException;
+import fi.nls.oskari.control.ActionException;
+import fi.nls.oskari.control.ActionParameters;
+import fi.nls.oskari.control.RestActionHandler;
+import fi.nls.oskari.domain.map.MaplayerGroup;
+import fi.nls.oskari.domain.map.OskariLayer;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.map.layer.OskariLayerService;
+import fi.nls.oskari.map.layer.OskariLayerServiceIbatisImpl;
+import fi.nls.oskari.util.IOHelper;
+import fi.nls.oskari.util.ResponseHelper;
+
+/**
+ * CRUD for layer and group order handling. Methods require admin user.
+ */
+@OskariActionRoute("LayerAndGroupOrder")
+public class LayerAndGroupOrderHandler extends RestActionHandler {
+	private static Logger log = LogFactory.getLogger(LayerAndGroupOrderHandler.class);
+	
+    private static final String KEY_PARENT = "parent";
+    private static final String KEY_ORDERS = "orders";
+    private static final String KEY_TYPE = "type";
+    private static final String KEY_ID = "id";
+    private static final String KEY_OLD_PARENT = "oldParent";
+    private static final String TYPE_LAYER = "layer";
+	
+	private OskariMapLayerGroupService oskariMapLayerGroupService;
+	private OskariLayerService oskariLayerService;
+	
+	/**
+	 * Variable to keep track of the new order of the groups.
+	 */
+	private HashMap<Integer, Integer> newGroupOrderMap = new HashMap<>();
```

Comment:
Is `newGroupOrderMap` or `newLayerOrderMap` used somewhere? I can't find any usage.